### ISSUE TITLE
0.17.20 on brew installs a binary which reports 0.17.19

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#2879]
From community slack: https://pulumi-community.slack.com/archives/C84L4E3N1/p1561558863347700

> Um, I think something is wrong with pulumi. `pulumi version` gives `v0.17.19` but `brew info pulumi` gives `v0.17.20`.

**Comment by lukehoban** [comment]
A user also reports brew errors on this release:

```
brew upgrade
==> Upgrading 1 outdated package:
pulumi 0.17.18 -> 0.17.20
==> Upgrading pulumi
==> Cloning https://github.com/pulumi/pulumi.git
Cloning into '/home/az/.cache/Homebrew/pulumi--git'...
Note: checking out 'f59a9340442b6c2b6390728ff7de60cf6bbc96ac'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

==> Checking out tag v0.17.20
HEAD is now at f59a934 Add EngineEventsBatch type (#2858)
Error: An exception occurred within a child process:
  RuntimeError: v0.17.20 tag should be 9823035de4bbbafe2df9080348a0b36665e26895
but is actually f59a9340442b6c2b6390728ff7de60cf6bbc96ac
```

**Comment by ellismg** [comment]
Two bad things happened during the release of `v0.17.20`:

1. We initially tagged the same commit twice, once as `v0.17.19` and once as `v0.17.20`.  This caused problems because our version scripts use the tags to figure out what version to apply and when faced with both tags they picked the "older" one.

2. We corrected our mistake later, tagging the correct commit as 0.17.20, and re-pushing.

Between (1) and (2), Homebrew contributors picked up 0.17.20 (a totally reasonable thing to do).  This meant the prebuilt binaries had the wrong version information in them, and later when we moved the tag around we screwed up their integrity checks.  This was a problem when building from source (but I don't believe would have affected installed that were using the prebuilt binaries).

We have now released 0.17.21, and [Homebrew has been updated](https://formulae.brew.sh/formula/pulumi) with bits that should be correct.

The major takeaway here is that in the future if we make the same mistake, we shouldn't try moving tags around, especially if homebrew has already picked up a version.  We should say: "Well, we screwed that up, let's bump the version and correctly tag things" instead of trying to move the wrong tag around.
